docs/blueprints/reputation-store.md

```diff
@@ -0,0 +1,107 @@
+// once a node becomes offline_suspended, mark it as under review so we check it again later
+field under_review timestamp ( updatable, nullable )
+field online_score float64 ( updatable, default 1 )
+field audit_history blob ( updatable )
+
+// audit_reputation_fields track information related to successful vs. failed error audits
+field audit_reputation_alpha  float64 ( updatable, default 1 )
+field audit_reputation_beta   float64 ( updatable, default 0 )
+// unknown_audit_reputation fields track information related to successful vs. unknown error audits
+field unknown_audit_reputation_alpha  float64 ( updatable, default 1 )
+field unknown_audit_reputation_beta   float64 ( updatable, default 0 )
+
+```
+
+### Other Details
+
+We will need to move the audit history logic out of the `overlay` package. We can add the changes discussed in this design to the `audit` package.
+
+If we decide to use Cockroach as the implementation for the store, we can repurpose the `audit_histories` table, as long as we make sure to remove references to it from queries in the overlay.
+
+## Rationale
+
+One solution to this problem that doesn't involve a store would be to create an in-memory cache tied to each service that needs to do audit updates (e.g. audit, repair). The cache would need to pull reputation information from the `nodes` table on startup, keep it up to date as audits occur, and flush on a time interval or whenever a contested field for a node changes due to audits (e.g. disqualified).
+
+We could still implement an in-memory cache with the design outlined above with a few key changes:
+* Add a `Initialize(nodeID storj.NodeID, reputation AllReputationFields)` function to the interface, allowing us to set the reputation for a node in the cache based on the `nodes` table on satellite startup.
+* Update the `NodeStatusChange` struct to include _all_ reputation values, not only the contested values. While a reputation store can keep track of the frequently-updated reputation values between restarts, a cache will need to update these values in the `nodes` table to preserve them.
+* Include a `last_flushed` field in the cache for _each_ node. Whenever `NodeStatusChange` is returned by `UpdateNode`, the caller is expected to transfer these changes over to the `nodes` table. The cache can return `NodeStatusChange` either when a contested column such as `disqualified` changes, or when enough time (e.g. 1 hour) has passed since `last_flushed` for this particular node.
+
+We could also implement a reputation cache as a completely separate satellite service. It could handle all the details of keeping track of reputation values, as well as flushing on an interval or when contested fields change. The necessity of having a separate package depends on the amount of code necessary to implement a solution. If, for example, we can simply use Cockroach, the code additions are minimal. But if we want an in-memory cache, we may need to have another package.
```

Comment:
I think it's going to be clearer if we have a separate service that talks to the nodes and reputation tables. So we don't actually expose NodeStatusChange externally from the package.